src/apps/tezos/control_baking.py

```diff
@@ -0,0 +1,29 @@
+from trezor import config
+from trezor.messages.Failure import Failure
+from trezor.messages.Success import Success
+from trezor.pin import pin_to_int
+
+from apps.common.request_pin import request_pin
+from apps.tezos import helpers, layout
+
+
+async def control_baking(ctx, msg):
+
+    if not config.has_pin():
+        await layout.no_pin_dialog(ctx)
```

Comment:
this should probably return Failure (or raise an error) straight away, see comments below